api/src/main/java/org/openmrs/module/idgen/SequentialIdentifierGenerator.java

```diff
@@ -187,4 +187,17 @@ public Long getNextSequenceValue() {
 	public void setNextSequenceValue(Long nextSequenceValue) {
 		this.nextSequenceValue = nextSequenceValue;
 	}
+	
+	private PrefixProvider getPrefixProvider(String providerName) {
+		return (PrefixProvider) IdgenUtil.getNewInstanceFromClassName(providerName);
```

Comment:
Using Spring beans could lead to something like that:
```java
private PrefixProvider getPrefixProvider(String prefixProviderBean) {
  return Context.getRegisteredComponent(prefixProviderBean, PrefixProvider.class);
}
```
This would allow us to keep a better control on how those beans are instantiated.
Careful this shouldn't but _could_ return `null`, so you'll have to adapt your code.